Heading/Heading.js

```diff
@@ -0,0 +1,147 @@
+/**
+ * Agate styled labeled Heading components and behaviors
+ *
+ * @example
+ * <Heading
+ *   size="large"
+ *   spacing="small"
+ * >
+ *   A Content Section Heading
+ * </Heading>
+ *
+ * @module agate/Heading
+ * @exports Heading
+ * @exports HeadingBase
+ * @exports HeadingDecorator
+ */
+
+import kind from '@enact/core/kind';
+import Pure from '@enact/ui/internal/Pure';
+import PropTypes from 'prop-types';
+import compose from 'ramda/src/compose';
+import UiHeading from '@enact/ui/Heading';
+
+import Skinnable from '../Skinnable';
+
+import componentCss from './Heading.module.less';
+
+/**
+ * A labeled Heading component.
+ *
+ * This component is most often not used directly but may be composed within another component as it
+ * is within [Heading]{@link agate/Heading.Heading}.
+ *
+ * @class HeadingBase
+ * @memberof agate/Heading
+ * @ui
+ * @public
+ */
+const HeadingBase = kind({
+	name: 'Heading',
+
+	propTypes: /** @lends agate/Heading.HeadingBase.prototype */ {
+		css: PropTypes.object,
+
+		/**
+		 * Set a custom color to be used by the Heading. Defaults to the current accent color if
+		 * none is specified.
+		 *
+		 * @type {String}
+		 * @default null
+		 * @public
+		 */
+		customColor: PropTypes.string,
+
+		/**
+		 * Adds a horizontal-rule (line) under the component
+		 *
+		 * @type {Boolean}
+		 * @public
+		 */
+		showLine: PropTypes.bool,
+
+		/**
+		 * The size of the spacing around the Heading.
+		 *
+		 * Allowed values include:
+		 * * `'auto'` - Value is based on the `size` prop for automatic usage.
+		 * * `'large'` - Specifically assign the `'large'` spacing.
+		 * * `'medium'` - Specifically assign the `'medium'` spacing.
+		 * * `'small'` - Specifically assign the `'small'` spacing.
+		 * * `'none'` - No spacing at all. Neighboring elements will directly touch the Heading.
+		 *
+		 * @type {('auto'|'title'|'large'|'medium'|'small'|'none')}
+		 * @default 'small'
+		 * @public
+		 */
+		spacing: PropTypes.oneOf(['auto', 'title', 'large', 'medium', 'small', 'none'])
+	},
+
+	defaultProps: {
+		spacing: 'small'
```

Comment:
Why not have the default be `auto`?

Reply:
That's a good question. It probably should be.